File: 프로그래머스/lv3/72413. 합승 택시 요금/합승 택시 요금.java
import java.util.*;

class Solution {
    public int solution(int n, int s, int a, int b, int[][] fares) {

        int[][] price = new int[n+1][n+1];
        for (int i=0;i<price.length;i++) {
            Arrays.fill(price[i],10000000);
            price[i][i] = 0;
        }
        
        for (int i=0;i<fares.length;i++) {
            int c = fares[i][0];
            int d = fares[i][1];
            int f = fares[i][2];
            
            price[c][d] = f;
            price[d][c] = f;
        }
        
        for (int i=1;i<=n;i++) {
            for (int j=1;j<=n;j++) {
                for (int k=1;k<=n;k++) {
                    price[j][k] = Math.min(price[j][k],price[j][i]+price[i][k]);
                }
            }
        }
        int answer = price[s][a] + price[s][b];
        
        for (int i=1;i<=n;i++) {
            answer = Math.min(answer,price[s][i] + price[i][a] + price[i][b]);
        }
        
        return answer;
    }
}

File: 프로그래머스/3/72413. 합승 택시 요금/합승 택시 요금.py
import math

def solution(n, s, a, b, fares):
    
    
    taxi_price = [[math.inf] * (n+1) for _ in range(n+1)]
    
    for first, second, fare in fares:
        taxi_price[first][second] = fare
        taxi_price[second][first] = fare 
        
    for i in range(1,n+1):
        taxi_price[i][i] = 0
    
    for k in range(1,n+1):
        for i in range(1,n+1):
            for j in range(1,n+1):
                taxi_price[i][j] = min(taxi_price[i][j], taxi_price[i][k] + taxi_price[k][j])
                
 
    answer = math.inf
    for i in range(1,n+1):
        answer = min(answer , taxi_price[s][i] + taxi_price[i][a] + taxi_price[i][b])
        
            
    return answer